src/mmrelay/meshtastic/connection.py

```
@@ -0,0 +1,1359 @@
+    try:
+        return facade._connect_meshtastic_impl(
+            passed_config=passed_config,
+            force_connect=force_connect,
+        )
+    finally:
+        with facade._connect_attempt_condition:
+            facade._connect_attempt_in_progress = False
+            facade._connect_attempt_condition.notify_all()
+
+
+def _connect_meshtastic_impl(
```

Ruff (0.15.9)

[warning] Too many return statements (20 > 6)

(PLR0911)